script/cache_critical_dns

```diff
@@ -23,6 +23,23 @@ CRITICAL_HOST_ENV_VARS = %w{
 HOST_RESOLVER_CACHE = {}
 HOST_HEALTHY_CACHE = {}
 HOSTS_PATH = ENV['DISCOURSE_DNS_CACHE_HOSTS_FILE'] || "/etc/hosts"
+
+PrioFilter = Struct.new(:le, :ge) do
+  # le and ge must be callable functions (e.g. a lambda) which take a single
+  # argument - the priority of an SRV RR target - and return a boolean value.
+  # The return value indicates if the priority is less than or equal to the
+  # upper threshold, or greater than or equal to the lower threshold, respectively.
+  # A target's priority is only within the threshold if both le(max prio) and
+  # ge(min prio) are true.
+  def within_threshold?(p)
+    le.call(p) && ge.call(p)
```

Comment:
Do `le` and `ge` need to be unary funcs?  Would this be simpler?

(Note I've inverted the order of `:le` and `:ge` in the constructor.)

```ruby
#!/usr/bin/env ruby

PrioFilter = Struct.new(:ge, :le) do
  def within_threshold?(p)
    p >= ge && p <= le
  end
end
SRV_PRIORITY_THRESHOLD_MIN = 0
SRV_PRIORITY_THRESHOLD_MAX = 65535
SRV_PRIORITY_FILTERS = Hash.new(PrioFilter.new(SRV_PRIORITY_THRESHOLD_MIN, SRV_PRIORITY_THRESHOLD_MAX))

# only accept maxstreme most preferred
SRV_PRIORITY_FILTERS["TEST_SRV"] = PrioFilter.new(0, 0)

puts(SRV_PRIORITY_FILTERS["hello"].within_threshold?(10))
puts(SRV_PRIORITY_FILTERS["TEST_SRV"].within_threshold?(10))
```

```
% ./dunnoruby
true
false
```

Reply:
Classic case of trying to be Very Clever. The lambdas were a leftover of the initial implementation, which didn't have the `within_threshold?` method and called the le, ge functions directly. 

Anyway, I've ripped that out and this is more concise.